Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -853,12 +853,13 @@ def line_problems(line, _lineno)
       # TODO: check could be in RuboCop
       problem "`env :userpaths` in formulae is deprecated" if line.include?("env :userpaths")
 
-      if line =~ /system ((["'])[^"' ]*(?:\s[^"' ]*)+\2)/
-        bad_system = Regexp.last_match(1)
-        unless %w[| < > & ; *].any? { |c| bad_system.include? c }
-          good_system = bad_system.gsub(" ", "\", \"")
+      if line =~ /(system|Utils\.(?:safe_)?popen(?:_read|_write)?) ((["'])[^"' ]*(?:\s[^"' ]*)+\3)/
```

Comment:
This regex is a bit impenetrable. It wasn't great before but it's pretty bad now. How about porting this whole check to RuboCop instead (as-per the comment below and https://github.com/Homebrew/brew/issues/7392)? That would allow it to actually parse rather than trying to match with a regex.

Reply:
Porting to RuboCop would be great!

[RegExper](https://regexper.com/#%28system%7CUtils%5C.%28%3F%3Asafe_%29%3Fpopen%28%3F%3A_read%7C_write%29%3F%29%20%28%28%5B%22'%5D%29%5B%5E%22'%20%5D*%28%3F%3A%5Cs%5B%5E%22'%20%5D*%29%2B%5C3%29) is fantastic for visualizing regular expressions. It's fantastic.

![image-2](https://user-images.githubusercontent.com/291551/83904416-ace38d80-a714-11ea-8848-bc245ce1c3a7.png)

Reply:
I sad fantastic twice. That's because it's doubly fantastic.

Reply:
@MikeMcQuaid Right now the way I'm doing it with RuboCop still uses a regex. Was there something different you have in mind? Here is an outline of the rubocop code:

```ruby
def audit_formula(_node, _class_node, _parent_class_node, body_node)
  find_every_method_call_by_name(body_node, :system).each do |method|
    next unless match = regex_match_group(parameters(method).first, /<regex goes here>/)

    problem "Use ENV instead of invoking '#{match[1]}' to modify the environment"
  end
end
```